src/hotspot/cpu/x86/stubRoutines_x86.hpp

```diff
@@ -38,7 +38,7 @@ enum platform_dependent_constants {
   // AVX512 intrinsics add more code in 64-bit VM,
   // Windows have more code to save/restore registers
   _compiler_stubs_code_size     = 20000 LP64_ONLY(+47000) WINDOWS_ONLY(+2000),
-  _final_stubs_code_size        = 10000 LP64_ONLY(+20000) WINDOWS_ONLY(+2000) ZGC_ONLY(+20000)
+  _final_stubs_code_size        = 10000 LP64_ONLY(+20000) WINDOWS_ONLY(+22000) ZGC_ONLY(+20000)
```

Comment:
Do you know which particular stubs cause out of space?
Most stubs are generated for `compiler` intrinsics. I would assume that they are causing the issue and not `final` 6 stubs.

Reply:
Scratch that. I see call stack in the bug report. It is indeed final stubs for upcall stub when ZGC is enabled.
Should we increase ZGC_ONLY(+20000) value instead?

Reply:
ZGC save/restores XMM registers regardless OS: [x86/gc/z/zBarrierSetAssembler_x86.cpp#L70](https://github.com/openjdk/jdk/blob/master/src/hotspot/cpu/x86/gc/z/zBarrierSetAssembler_x86.cpp#L70)

Reply:
I spent a while reading the dumps of final stubs. I can account for 10k additional for XMM stubs in final on Windows. This takes it to the edge.

It may well be that this is really a ZGC thing, but with ZGC on Linux we get nowhere near the limit. I think there is about 20k difference between Linux and Windows with ZGC enabled. I can't account for all of it, though.

Reply:
Okay, thank you for details.

Reply:
FTR,
Following are the final stub section sizes with and without -XX:+UseZGC on linux and windows

Linux
 - With ZGC
    [0.928s][info][stubs] StubRoutines (final stubs)         [0x00007fb1afc320c0, 0x00007fb1afc3e550] used: 41659, free: 8661
 - Without ZGC
 -  [0.019s][info][stubs] StubRoutines (final stubs)         [0x00007f5e6fc2cc40, 0x00007f5e6fc390d0] used: 27419, free: 22901

Windows:
   - With ZGC 
   -  [0.043s][info][stubs] StubRoutines (final stubs)         [0x00000291ea5684c0, 0x00000291ea575120] used: 45870, free: 6450
   - Without ZGC
   - [0.039s][info][stubs] StubRoutines (final stubs)         [0x000002003a1e31c0, 0x000002003a1efe20] used: 30030, free: 22290